### PR TITLE
wallet-new:  run EKG server if --ekg-server IP:PORT is passed

### PR DESCRIPTION
**Pull request by deepfire**
This allows the new wallet to start the EKG server if `--ekg-server IP:PORT` is passed.

## Type of change

- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🛠 New feature (non-breaking change which adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🏭 Refactoring that does not change existing functionality but does improve things like code readability, structure etc
- [ ] 🔨 New or improved tests for existing code
- [ ] ⛑ git-flow chore (backport, hotfix, etc)

## Developer checklist

- [ ] I have read the [style guide](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/style-guide.md) document, and my code follows the code style of this project.
- [ ] If my code deals with exceptions, it follows the [guidelines](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/exceptions.md).
- [ ] I have updated any documentation accordingly, if needed. Documentation changes can be reflected in opening a PR on [cardanodocs.com](https://github.com/input-output-hk/cardanodocs.com), amending the inline [Haddock](https://www.haskell.org/haddock/) comments, any relevant README file or one of the document listed in the [docs](https://github.com/input-output-hk/cardano-sl/tree/develop/docs) directory.
- [ ] CHANGELOG entry has been added and is linked to the correct PR on GitHub.

## Testing checklist

- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

## How to merge

Send the message `bors r+` to merge this PR. For more information, see
[`docs/how-to/bors.md`](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/how-to/bors.md).

### CODE REVIEW

**Comment by amias-iohk** [comment]
i thought this was an experiment not a release feature , not sure why it should be merged to release

**Comment by deepfire** [comment]
@amias-iohk, this is a useful feature to have, going forward.

File: wallet-new/server/Main.hs [hunk 1]
```diff
@@ -11,14 +12,12 @@ import qualified Cardano.Wallet.LegacyAction as Legacy
...
 
-
 -- | The main entrypoint for the Wallet.
 main :: IO ()
 main = withCompileInfo $ do
     WalletStartupOptions cArgs wArgs <- getWalletNodeOptions
     let lArgs = loggingParams "node" cArgs
     let nArgs = NodeArgs { behaviorConfigPath = Nothing }
-
     putText "Wallet is starting..."
```
Comment (KtorZ) [inline wallet-new/server/Main.hs]: No big deal, but let's leave it untouch. I can already foresee the very confusing git merge conflicts incoming here :upside_down_face:

**Comment by amias-iohk** [comment]
i'm just concerned that it will alter behaviour of the code and introduce potential security issues.
I am also surprised that we are pushing features to release like this , didn't this cause lots of problems last time ? Remember that there will be limited QA resources in the next few weeks.

File: lib/src/Pos/Launcher/Launcher.hs [hunk 2]
```diff
@@ -63,6 +66,13 @@ launchNode nArgs cArgs lArgs action = do
             (cnaDumpConfiguration cArgs)
             confOpts
 
+    case ekgParams cArgs of
+      Just (EkgParams eHost ePort) -> do
+        putText $ "EKG is starting on " <> show eHost <> ":" <> show ePort
+        _ <- forkServer eHost ePort
+        pure ()
+      Nothing -> pure ()
```
Comment (KtorZ) [inline lib/src/Pos/Launcher/Launcher.hs]: Thanks @deepfire, this is better place for it :+1: 

Side-note: `universum` defines a helper `whenJust` for exactly this kind of case:

```hs
whenJust cArgs $ \(EkgParams eHost ePort) -> do
  putText $ "EKG is starting on " <> show eHost <> ":" <> show ePort
  void $ forkServer eHost ePort
```

**Comment by deepfire** [comment]
@amias-iohk, we aren't really adding code here -- we're just exposing what we already had.

It can't cause security issues, unless it's started -- and it won't be started on production wallets, certainly not by default.

File: wallet-new/server/Main.hs (hunk 1 above)
Comment (deepfire) [reply wallet-new/server/Main.hs]: Oh, right, sorry that slipped through!

File: lib/src/Pos/Launcher/Launcher.hs (hunk 2 above)
Comment (deepfire) [reply lib/src/Pos/Launcher/Launcher.hs]: Done!

**Comment by deepfire** [comment]
The remaining checks are all Darwin -- and we don't expect any divergence between OSes.

**Comment by deepfire** [comment]
Merging, since we want to expedite the release-critical debugging effort.

**Pull request merged**